src/main/java/edu/harvard/iq/dataverse/DataFileServiceBean.java

```diff
@@ -1657,4 +1689,29 @@ public Embargo findEmbargo(Long id) {
         DataFile d = find(id);
         return d.getEmbargo();
     }
+    
+    public Long getStorageUsageByCreator(AuthenticatedUser user) {
+        Query query = em.createQuery("SELECT SUM(o.filesize) FROM DataFile o WHERE o.creator.id=:creatorId");
+        
+        try {
+            Long totalSize = (Long)query.setParameter("creatorId", user.getId()).getSingleResult();
+            logger.info("total size for user: "+totalSize);
+            return totalSize == null ? 0L : totalSize; 
```

Comment:
when would this be null (as opposed to the exception thrown)? is is where we for some reason don't have sizes stored?